File: app_custom_training/cartridges/app_custom_training/cartridge/scripts/Data/ContainerWithMostWater.js
'use strict';


function containsWithWater(height) {
    var max = 0;
    var l = 0;
    var r = height.length - 1;
    var checks = 0;
    while (l < r) {
        checks++;
        var lh = height[l];
        var rh = height[r];
        var min_h = Math.min(lh, rh);
        max = Math.max(max, min_h * (r - 1))
        if (lh < rh) {
            l++;
        } else {
            r--;
        }
    }

    return max;

}

module.exports.containsWithWater = containsWithWater;

File: app_custom_training/cartridges/app_custom_training/cartridge/scripts/Data/containerwithMostWater.java
public int maxArea(int[] height) {
    int n = height.length;
    int maxArea = Integer.MIN_VALUE;
    int left = 1;
    int right = n;
    int h = Integer.MAX_VALUE;
    while ( left < right){
        int width = right - left;
        h = Math.min(height[left-1] , height[right-1]);
        int r = h * width;
        if(maxArea < r) {
            maxArea = r;
        }
        if(height[left-1] < height[right-1]){
            left++; //moved this line which you wrote earler to this if block
        }else {
            right--;
        }
    }
    return maxArea;
}
}